docs/decisions/0054-processes.md

```diff
@@ -0,0 +1,317 @@
+
+### Options #2:
+
+**_Build SK Process library within an existing workflow framework_**:
+Of all the frameworks explored, the few that seem closest to meeting the technical requirements listed above are based on [Durable Tasks](https://github.com/Azure/durabletask). This includes things like Dapr Workflows, Azure Durable Functions, or the Durable Tasks Framework itself. Attempts to build a working solution on these frameworks resulted an awkward interface for basic scenarios due to the underlying structure of Durable Tasks where nodes are stateless and only the central orchestrator is stateful. While it is likely that many AI driven workflows could be modeled in this type of system, our exploration did not produce something we were happy with from a usability perspective.
+
+### Options #3:
+
+**_Build SK Process library with a custom build workflow engine_**:
+Building a custom workflow engine might provide the cleanest integration but would require extensive resources and time that we don't have. Distributed workflow engines are products in and of themselves.
+
+### Options #4:
+
+**_Build platform agnostic SK Process library with connectors for existing workflow frameworks_**:
+This is the chosen option.
+
+## Decision Outcome
+
+**_Chosen option - #4_**: Build platform agnostic SK Process library with connectors for existing workflow frameworks.
+This was the only option that was ale to meet all the technical and scenario driven requirements. This option should allow for a simple and well-integrated interface into Semantic Kernel as well as the ability to support many existing distributed runtimes that will give our customers the flexibility to use their existing infrastructure and expertise.
+
+### Components of the Process library
+
+The proposed architecture of a Process is based on a graph execution model where nodes, which we call Steps, perform work by invoking user defined Kernel Functions. Edges in the graph are defined from an event driven perspective and carry metadata about the event as well as a data payload containing the output of the Kernel Function invocation.
+
+Starting from the ground up, the components of a processes are:
+
+1.  **_KernelFunctions_**: The same KernelFunctions that our customers already know and use. Nothing new here.
+1.  **_Steps_**: Steps group one ore more KernelFunctions together into an object with optional user defined state. A step represents one unit of work within a process. Steps make the output of their work visible to other steps in the process by emitting events. This event based structure allows steps to be created without needing to know which process they are used in, allowing them to be reusable across multiple processes.
```

Comment:
**What about non-deterministic data (ai generated) that comes as input of a step?** 
Maybe it would be worth adding a validator of some kind that verifies that the data is compliant with the "business rules"?

Ie, checking that the:
- text generated has the intended meaning and purpose and does not include jailbreaking intents.
- json generated has the intended structure, proper data...
- validate strongly typed inputs

Maybe implement a **middleware pattern** where we have some custom implemented validator middleware implemented ready to use classes - and is extensible, ideally (love extensibility, sorry about that...)